src/dialogs/services/dialogData.ts

```diff
@@ -0,0 +1,133 @@
+import * as _ from 'lodash';
+import * as angular from 'angular';
+
+export default class DialogDataService {
+  /**
+   * Sets up and configures properties for a dialog field
+   * @memberof DialogDataService
+   * @function setData
+   * @param data {any} This is a object that is all the information for a particular dialog field
+   *
+   **/
+  public setupField(data: any) {
+    let field = _.cloneDeep(data);
+    field.fieldValidation = null;
+    field.fieldBeingRefreshed = false;
+    field.errorMessage = '';
+    if (field.type === 'DialogFieldDropDownList') {
+      field.values = this.updateFieldSortOrder(field);
+    }
+    field.default_value = this.setDefaultValue(field);
+
+    if (field.type === 'DialogFieldDropDownList') {
+      for (let option of field.values) {
+        if (option[0] === String(field.default_value)) {
+          field.selected = option;
+        }
+      }
+    }
+
+    return field;
+  }
+  /**
+   *
+   * This method updates sort order of dialog options for a dialog field that is a drop down.
+   * @memberof DialogDataService
+   * @function updateFieldSortOrder
+   * @param data {any} This is a object that is all the information for a particular dialog field
+   *
+   **/
+  private updateFieldSortOrder(data) {
+    const values: any = data.values;
```

Comment:
Good feedback.  Some of these functions are carryovers from the existing components, this is a solid enhancement